src/Hybrid/transitions.jl

```diff
@@ -0,0 +1,179 @@
+end
+
+apply(f::DiscreteTransition,
+      R::AbstractLazyReachSet,
+      method::AbstractTMDisjointnessMethod) = _apply(f, set(R), method)
+
+# ==========================================
+# Cases where the reset map is the identity
+# ==========================================
+
+# ---------------------------
+# Intersection method: Exact
+# ---------------------------
+
+# all sets are polyhedral
+function _apply(tr::DiscreteTransition{<:Universe, <:ZeroSet, GT, IT⁻, IT⁺},
+                X::AbstractPolyhedron{N},
+                ::Exact) where {N,
+                                GT<:AbstractPolyhedron{N},
+                                IT⁻<:AbstractPolyhedron{N},
+                                IT⁺<:AbstractPolyhedron{N}}
+
+    clist_G = constraints_list(tr.G)
+    clist_I⁻ = constraints_list(tr.I⁻)
+    clist_I⁺ = constraints_list(tr.I⁺)
+    clist_X = constraints_list(X)
+    out = vcat(clist_G, clist_I⁻, clist_I⁺, clist_X)
+    flag = remove_redundant_constraints!(out)
+    if flag
+        return HPolyhedron(out)
+    else
+        return EmptySet(dim(X))
+    end
+end
+
+# the source invariant is the set union of half-spaces, the other sets are polyhedral
+function _apply(tr::DiscreteTransition{<:Universe, <:ZeroSet, GT, IT⁻, IT⁺},
+                X::AbstractPolyhedron{N},
+                ::Exact) where {N,
+                                GT<:AbstractPolyhedron{N},
+                                VN,
+                                IT⁻<:UnionSetArray{N, LinearConstraint{N, VN}},
+                                IT⁺<:AbstractPolyhedron{N}}
+
+    clist_G = constraints_list(tr.G)
+    clist_I⁺ = constraints_list(tr.I⁺)
+    clist_X = constraints_list(X)
+    aux = vcat(clist_G, clist_I⁺, clist_X)
+    flag = remove_redundant_constraints!(aux)
+    !flag && return EmptySet(dim(X))
+
+    clist_I⁻ = tr.I⁻.array
+    out = Vector{HPolyhedron{N, Vector{N}}}()
+
+    for ci in clist_I⁻
+        aux2 = vcat(aux, ci)
+        flag = remove_redundant_constraints!(aux2)
+        if !flag
+            continue
+        else
+            push!(out, HPolyhedron(copy(aux2)))
+        end
+    end
+
+    if isempty(out)
+        return EmptySet(dim(X))
+    else
+        return UnionSetArray(out)
+    end
+end
+
+# the source invariant is the set union of half-spaces, the other sets are polyhedral
+function _apply(tr::DiscreteTransition{<:Universe, <:ZeroSet, GT, IT⁻, IT⁺},
+                X::ConvexHullArray{N, PT},
+                ::Exact) where {N,
+                                PT<:AbstractPolyhedron{N},
+                                GT<:AbstractPolyhedron{N},
+                                IT⁻<:AbstractPolyhedron{N},
+                                IT⁺<:AbstractPolyhedron{N}}
+
+    vlist = [vertices_list(Xi) for Xi in array(X)]
+    vlist_ch = convex_hull(vcat(vlist...)) # TODO pass the backend as an option
+    Xhrep = tohrep(vlist_ch)
```

Comment:
I think this should be sent to `LazySets` as a first implementation of `constraints_list(X)`.